### PR TITLE
UI: read & write visual graph properties bugfix [ES-1078]

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

TLDR: Fixed reading and writing of visual graph properties (UI).
**Important:** This needs to be tested intensively, as we do not have automated UI tests (?) for this.

It seems that the response of (undocumented?) API `_api/user/<user>/config` has changed. 
* First, results are not handled at top-level anymore (e.g. now it is `results.graphs` instead of `graphs`).
* Second, it seems that also the modification logic changed internally. Beforehand it used to be a `PATCH` internally,
but now it is a `REPLACE`. I am pretty sure that this behaviour has changed. As this whole graph storage of properties worked once. 

Nevertheless, this PR should fix the behaviour.
Still there is a bit of future work todo regarding code read- & maintainability.

- [x] :hankey: Bugfix

### Checklist

- [x] :book: CHANGELOG entry made
- [x] Backports: Yes Backports are required!
  - [x] Backport for 3.10: #17443

#### Related Information

- [x] Enterprise PR: No enterprise PR required.
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/ES-1078

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/frontend/js/models/userConfig.js [hunk 1]
```diff
@@ -92,32 +92,91 @@ window.UserConfig = Backbone.Model.extend({
...
+        return newObject;
+      }
+
+      // Format notice
+      // dataFromDB<object>: dataFromDB.result.keyName
+      if (dataFromDB.result && dataFromDB.result[keyName]) {
+        const existingEntry = dataFromDB.result[keyName];
+        // We've found available data we need to merge
+        if (isObject(existingEntry) && isObject(newObject)) {
+          return {...existingEntry, ...newObject};
```
Comment (adityamukho) [inline js/apps/system/_admin/aardvark/APP/frontend/js/models/userConfig.js:140]: This is a shallow merge - hope it is sufficient for this use case?

**Review by adityamukho — COMMENTED** [review]
IMO the ideal place to store custom user-specific data is in the `extra` object of the user's profile. This is read by `GET _api/user/{user}` and is patchable (by first merging the extra object with new data on the client side) using `PATCH _api/user/{user}`.

The benefit is that this endpoint is documented.

Please see my other comments inline.

File: js/apps/system/_admin/aardvark/APP/frontend/js/models/userConfig.js (hunk 1 above)
Comment (ajurdak) [reply js/apps/system/_admin/aardvark/APP/frontend/js/models/userConfig.js:140]: After research in our specific case this solution is okay as we always save the whole object and never only change one specific property.

**Review by ajurdak — APPROVED** [review]
LGTM! I tested the functionality a lot.

**Review by nikhil-varma — APPROVED** [review]

**Review by adityamukho — APPROVED** [review]
LGTM
